src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/ciagmigration/WebClientConfiguration.kt

```diff
@@ -0,0 +1,47 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.ciagmigration
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.http.client.reactive.ReactorClientHttpConnector
+import org.springframework.security.oauth2.client.AuthorizedClientServiceOAuth2AuthorizedClientManager
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientManager
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientProviderBuilder
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientService
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository
+import org.springframework.security.oauth2.client.web.reactive.function.client.ServletOAuth2AuthorizedClientExchangeFilterFunction
+import org.springframework.web.reactive.function.client.WebClient
+import reactor.netty.http.client.HttpClient
+import kotlin.apply as kotlinApply
+
+@Configuration
+class WebClientConfiguration(
```

Comment:
This config class, and the beans within it (specifically `authorizedClientManager` and the extension function `authorisedWebClient`) are largely modelled on (copy and paste) the same class from `prisoner-search` which I was pointed at as an example of how to do service to service API calls.

https://github.com/ministryofjustice/hmpps-prisoner-search/blob/main/hmpps-prisoner-search-indexer/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/config/WebClientConfiguration.kt

https://mojdt.slack.com/archives/C012FNNQ4A3/p1700152322411019